core/src/services/gcs/backend.rs

```diff
@@ -410,16 +411,32 @@ impl Accessor for GcsBackend {
     }
 
     async fn write(&self, path: &str, args: OpWrite) -> Result<(RpWrite, Self::Writer)> {
-        if args.append() {
-            return Err(Error::new(
-                ErrorKind::Unsupported,
-                "append write is not supported",
-            ));
-        }
+        let upload_location = if args.append() {
+            let resp = self.core.gcs_initiate_resumable_upload(path).await?;
+            let status = resp.status();
+
+            match status {
+                StatusCode::OK => {
+                    let bs = resp
```

Comment:
We can use [parse_location](https://github.com/apache/incubator-opendal/blob/main/core/src/raw/http_util/header.rs#L45)